src/cleanfigure.m

```diff
@@ -449,6 +451,47 @@ function simplifyLine(meta, handle, targetResolution)
     removeData(meta, handle, id_remove);
 end
 % =========================================================================
+function simplifyStairs(meta, handle)
+   % Some data might not lead to a new step in the stair, e.g.
+   % [(x_1, y_1), (x_2, y_1), ... (x_k, y_1), (x_{k+1} y_2)] 
+   % equals [(x_1, y_1), (x_{k+1} y_2)]
+
+    % Extract the data
+    xData = get(handle, 'XData');
+    yData = get(handle, 'YData');
+
+    % Dont do anything if the data is empty
+    if isempty(xData) || isempty(yData)
+        return;
+    end
+
+    % Check for identic data points
+    xNoDiff      = [false, (diff(xData) == 0)];
+    yNoDiff      = [false, (diff(yData) == 0)];
+
+    % Do not remove the last data point
+    xNoDiff(end) = false;
+    yNoDiff(end) = false;
+
+    % Identic data might lead to a visible result if the sign in the other
+    % dimension changes , e.g. the sequence  [0, 1], [0, -1], [0, 2]
+    % Therefore check for a sign change in the difference between data
+    % points (The sign changes if diff(sign)~=0)
+    xIsMonotone  = [true, diff(sign(diff(xData)))==0, true];
```

Comment:
I agree that we don't need global monotonicity to able to simplify the graph.
Sorry, I had a long day yesterday, so I think my reasoning skills were not ideal.